lib/optimizely/event_builder.rb

```diff
@@ -41,198 +42,202 @@ def ==(event)
+        experiment_id: experiment_id,
+        variation_id: variation_id,
+      }]
+
+      event = [{
+        entity_id: @config.experiment_key_map[experiment_key]['layerId'],
+        timestamp: get_timestamp(),
+        key: ACTIVATE_EVENT_KEY,
+        uuid: get_uuid()
+      }]
+
+      impressionEventParams = {
+        decisions: decision,
+        events: event
+      }
 
-    def add_user_id(user_id)
-      @params['visitorId'] = user_id
+      impressionEventParams;
     end
 
-    def add_attributes(attributes)
-      @params['userFeatures'] = []
-
-      return if attributes.nil?
+    def get_conversion_event_params(event_key, event_tags, experiment_variation_map)
+      # Creates object of params specific to conversion events
+      #
+      # event_key -                +String+ Event key representing the event which needs to be recorded
+      # event_tags -               +Hash+ Values associated with the event.
+      # experiment_variation_map - +Hash+ Map of experiment IDs to bucketed variation IDs
+      #
+      # Returns +Hash+ Impression event params
 
-      attributes.keys.each do |attribute_key|
-        # Omit falsy attribute values
-        attribute_value = attributes[attribute_key]
-        next unless attribute_value
+      conversionEventParams = []
 
-        # Skip attributes not in the datafile
-        attribute_id = @config.get_attribute_id(attribute_key)
-        next unless attribute_id
+      experiment_variation_map.each do |experiment_id, variation_id|
 
-        feature = {
-          'id' => attribute_id,
-          'name' => attribute_key,
-          'type' => 'custom',
-          'value' => attribute_value,
-          'shouldIndex' => true,
+        decision = {
+          decisions: [{
+            campaign_id: @config.experiment_id_map[experiment_id]['layerId'],
+            experiment_id: experiment_id,
+            variation_id: variation_id,
+          }],
+          events: [],
         }
-        @params['userFeatures'].push(feature)
-      end
-    end
-
-    def add_decision(experiment, variation_id)
-      experiment_key = experiment['key']
-      experiment_id = experiment['id']
-      @params['layerId'] = @config.experiment_key_map[experiment_key]['layerId']
-      @params['decision'] = {
-        'variationId' => variation_id,
-        'experimentId' => experiment_id,
-        'isLayerHoldback' => false,
-      }
-    end
 
-    def add_event_tags(event_tags)
-      @params['eventFeatures'] ||= []
-      @params['eventMetrics'] ||= []
+        event_object = {
+          entity_id: @config.event_key_map[event_key]['id'],
+          timestamp: get_timestamp(),
+          uuid: get_uuid(),
+          key: event_key,
+        }
 
-      return if event_tags.nil?
+        if event_tags
+          event_revenue = Helpers::EventTagUtils.get_revenue_value(event_tags)
+          if event_revenue
+            event_object[:revenue] = event_revenue
```

Comment:
can't we take it as a constant :revenue or what does it mean? Sorry for lack of knowledgein ruby. :)

Reply:
dictionary[:key] = value , this adds a property to an object. The event_object after this line will have 
'revenue' => value inside(event_revenue) defined

Reply:
Ruby have symbols data types and it is used for keys in Hash data type (Hash are similar to  array with indexes)